### PR TITLE
Update plugin header to use new plugin requirement values.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #5076

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: google-site-kit.php [hunk 1]
```diff
@@ -8,15 +8,17 @@
...
- * Author:      Google
- * Author URI:  https://opensource.google.com
- * License:     Apache License 2.0
- * License URI: https://www.apache.org/licenses/LICENSE-2.0
- * Text Domain: google-site-kit
+ * Plugin Name:    Site Kit by Google
+ * Plugin URI:     https://sitekit.withgoogle.com
+ * Description:    Site Kit is a one-stop solution for WordPress users to use everything Google has to offer to make them successful on the web.
+ * Version:        1.73.0
+ * Requires up to: 4.7
```
Comment (aaemnnosttv) [inline google-site-kit.php:15]: After closer inspection, this was defined incorrectly in the ACs (my mistake!).

This should be `Requires at least`, just like in `readme.txt`. I confused it with "Tested up to".

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Sorry, I noticed there was a mistake in the ACs after testing this.

In PHP this can be tested by calling
```php
get_plugin_data( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
```

Which should give a result like so
```
=> array(14) {
  ["Name"]=>
  string(18) "Site Kit by Google"
  ["PluginURI"]=>
  string(30) "https://sitekit.withgoogle.com"
  ["Version"]=>
  string(6) "1.73.0"
  ["Description"]=>
  string(193) "Site Kit is a one-stop solution for WordPress users to use everything Google has to offer to make them successful on the web. <cite>By <a href="https://opensource.google.com">Google</a>.</cite>"
  ["Author"]=>
  string(50) "<a href="https://opensource.google.com">Google</a>"
  ["AuthorURI"]=>
  string(29) "https://opensource.google.com"
  ["TextDomain"]=>
  string(15) "google-site-kit"
  ["DomainPath"]=>
  string(0) ""
  ["Network"]=>
  bool(false)
  ["RequiresWP"]=>
  string(3) "4.7"
  ["RequiresPHP"]=>
  string(3) "5.6"
  ["UpdateURI"]=>
  string(0) ""
  ["Title"]=>
  string(63) "<a href="https://sitekit.withgoogle.com">Site Kit by Google</a>"
  ["AuthorName"]=>
  string(6) "Google"
}
```

Note the corresponding values for `RequiresWP` and `RequiresPHP`.

**Comment by tofumatt** [comment]
Oh, thanks for the bit about `get_plugin_data`! 👍🏻  In that case; I added a very basic test 🙂

File: tests/phpunit/integration/PluginTest.php [hunk 2]
```diff
@@ -43,6 +43,14 @@ public function test_context() {
 		$this->assertEquals( plugin_dir_path( GOOGLESITEKIT_PLUGIN_MAIN_FILE ), $plugin->context()->path() );
 	}
 
+	public function test_plugin_data() {
+		// Ensure we're supplying the correct minimum version of PHP
+		// and WordPress in our plugin file's header.
+		$this->assertEquals( get_plugin_data( GOOGLESITEKIT_PLUGIN_MAIN_FILE )['RequiresPHP'], '5.6' );
```
Comment (aaemnnosttv) [inline tests/phpunit/integration/PluginTest.php:49]: Unfortunately, `RequiresPHP` wasn't added until 5.3 so this fails on our job using the minimum versions of PHP and WP.

https://github.com/google/site-kit-wp/runs/6260212583?check_suite_focus=true#step:10:44

Let's conditionally make this assertion based on the current WP version.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @tofumatt – this looks good now and thanks for adding the test! 

Just one thing left to fix 🙂

File: tests/phpunit/integration/PluginTest.php (hunk 2 above)
Comment (tofumatt) [reply tests/phpunit/integration/PluginTest.php]: Argh, whoops, I only tested locally with PHP 7. Will-do, thanks! 👍🏻

File: tests/phpunit/integration/PluginTest.php [hunk 3]
```diff
@@ -43,10 +43,18 @@ public function test_context() {
...
+	// Ensure we're supplying the correct minimum version of PHP
+	// and WordPress in our plugin file's header.
 	public function test_plugin_data() {
-		// Ensure we're supplying the correct minimum version of PHP
-		// and WordPress in our plugin file's header.
-		$this->assertEquals( get_plugin_data( GOOGLESITEKIT_PLUGIN_MAIN_FILE )['RequiresPHP'], '5.6' );
+		// RequiresPHP is only available in WordPress 5.3+, so make
+		// sure we're using that version before checking for that field
+		// in the plugin data.
+		if ( floatval( get_bloginfo( 'version' ) ) >= 5.3 ) {
```
Comment (aaemnnosttv) [inline tests/phpunit/integration/PluginTest.php:52]: PHP has a function for comparing versions which we should use instead.

**Review by aaemnnosttv — COMMENTED** [review]
The PHP test is still failing on WP 4.7 but I think I see why. I'll push a quick change and then this should be good to go.

**Comment by aaemnnosttv** [comment]
Turns out the header for `RequiresWP` also doesn't exist before 5.3 so this is now checked conditionally as well. Should be good to go now 👍

**Review by aaemnnosttv — APPROVED** [review]

**Pull request merged**